### PR TITLE
Add TagKeys and TagValues methods to Storage service

### PR DESCRIPTION
**Pull request by desa**
Link #13251

TODO:

- [x] Run `go generate ./storage/...`

_Briefly describe your proposed changes:_

This PR extends the `Storage` service interface by adding two methods `TagKeys` and `TagValues`. Both methods share request and response structures as they are particularly similar. ~The `TagsResponse` message shares a structure similar to that of the `ReadResponse`, but without all of the other frame types.~

  - [ ] Rebased/mergeable
  - [ ] Tests pass

### CODE REVIEW

File: storage/reads/datatypes/storage_common.proto [hunk 1]
```diff
@@ -208,6 +214,22 @@ message TimestampRange {
...
+  Predicate predicate = 3;
+}
+
+// Response message for Storage.TagKeys and Storage.TagValues.
+message TagsResponse {
+  message Frame {
+    repeated bytes values = 1;
+  }
+
+  repeated Frame frames = 1 [(gogoproto.nullable) = false];
```
Comment (stuartcarnie) [inline storage/reads/datatypes/storage_common.proto]: nit: I don't believe we'll need this inner `Frame` message; we can probably just include the values property in the `TagsResponse`.

Comment (desa) [reply storage/reads/datatypes/storage_common.proto]: 👍

**Comment by jlapacik** [comment]
Should a request for tag values include an explicit field for the tag key(s) being queried?

**Comment by stuartcarnie** [comment]
> Should a request for tag values include an explicit field for the tag key(s) being queried?

@jlapacik per the original RFC, Edd proposed the following API:

```go
// TagValues returns an iterator of tag values for the bucket matching predicate within the time range (start, end].
func (e *Engine) TagValues(orgID, bucketID influxdb.ID, start, end int64, predicate influxql.Expr) StringIterator {
    // This method would be invoked when the consumer wants to get the following schema information for an arbitrary
    // time range in a single bucket:
    // 
    // * All measurement names;
    // * All field keys;
    // * All tag values filtered by an arbitrary predicate, e.g., tag values for a measurement and tag key.
    //
}
```

It does not mention a tag key for which set of tag values to produce. @desa has undoubtedly based his definition from this. I suspect we will need to specify a tag key. It technically could be expressed in the predicate with a sentinel like `TAGKEY == "foo_key"`, which is similar to the old metadata queries

**Comment by desa** [comment]
Closing PR in favor of https://github.com/influxdata/influxdb/pull/13263

**Pull request closed without merging**